app/src/main/java/org/jellyfin/mobile/utils/SystemUtils.kt

```diff
@@ -39,35 +41,51 @@ fun MainActivity.requestNoBatteryOptimizations() {
 }
 
 fun MainActivity.requestDownload(uri: Uri, title: String, filename: String) {
-    val request = DownloadManager.Request(uri)
-        .setTitle(title)
-        .setDescription(getString(R.string.downloading))
-        .setDestinationInExternalPublicDir(Environment.DIRECTORY_DOWNLOADS, filename)
-        .setNotificationVisibility(DownloadManager.Request.VISIBILITY_VISIBLE_NOTIFY_COMPLETED)
-    val downloadMethod = appPreferences.downloadMethod
-    if (downloadMethod >= 0) {
-        downloadFile(request, downloadMethod)
-    } else runOnUiThread {
-        AlertDialog.Builder(this)
-            .setTitle(R.string.network_title)
-            .setMessage(R.string.network_message)
-            .setNegativeButton(R.string.wifi_only) { _, _ ->
-                val selectedDownloadMethod = DownloadMethod.WIFI_ONLY
-                appPreferences.downloadMethod = selectedDownloadMethod
-                downloadFile(request, selectedDownloadMethod)
-            }
-            .setPositiveButton(R.string.mobile_data) { _, _ ->
-                val selectedDownloadMethod = DownloadMethod.MOBILE_DATA
-                appPreferences.downloadMethod = selectedDownloadMethod
-                downloadFile(request, selectedDownloadMethod)
-            }
-            .setPositiveButton(R.string.mobile_data_and_roaming) { _, _ ->
-                val selectedDownloadMethod = DownloadMethod.MOBILE_AND_ROAMING
-                appPreferences.downloadMethod = selectedDownloadMethod
-                downloadFile(request, selectedDownloadMethod)
+    val requestWritePermissions = Build.VERSION.SDK_INT <= 28
+
+    fun initializeDownload() {
```

Comment:
A function in a function? :confused:
I'd actually prefer coroutines in that case..

Reply:
I had to put it somewhere haha